ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -142,7 +142,7 @@ const ConfirmPageContainer = (props) => {
       fromAddress,
       global.ethereumProvider,
     );
-    setCollectionBalance(tokenBalance?.balance?.words?.[0] || 0);
+    setCollectionBalance(tokenBalance.toString(10) || '0');
```

Comment:
Would be very very helpful to add some type of test for this...

Reply:
@segun The test for changes in this PR will be covered in a separate PR #19149  @digiwand  is working on, which will require some clean upon for the confirm-page-container test file as well.

Reply:
ok